src/blog/2024/05/node-red-mind-stack-with-flowfuse.md

```diff
@@ -0,0 +1,67 @@
+---
+title: The MIND stack with Node-RED and FlowFuse Dashboard 2.0
+subtitle: Leveraging FlowFuse for a Lighter MIND Stack (MIND = MQTT, InfluxDB, Node-RED, Dashboard 2.0)
+description: Our objective is to streamline the deployment of the MIND stack, enhancing its usability without compromising functionality.
+date: 2024-05-07
+authors: ["grey-dziuba", "harshad-joshi"]
+image: /blog/2024/05/images/mind-mqtt-influxdb-node-red-dashboard-2-0.png
+tags:
+    - posts
+    - flowfuse
+    - mind
+    - ming
+    - influx
+    - dashboards
+---
+
+The MING stack has gained significant popularity over the years as it built upon open-sourced projects.  That has given way to many people leveraging this stack to build solutions upon in various different environments.  The MING stack is composed of 4 main components, MQTT, InfluxDB, Node-RED, and Graphana.  Combined together are the 4 main pillars, data transportation, data storage, data transformation, and visualizations.  With this, it requires the management of 4 different applications, which often reside on the same server, but not necessarily.  With more moving parts, creates complexity.
+
+<!--more-->
+
+## FlowFuse: Integrated IoT Deployment
+
+With FlowFuse we have bundled up 3 of those 4.  First and foremost, the FlowFuse platform deploys Node-RED in two main ways.  A cloud based deployment and remote deployment with management from the FlowFuse platform.  Part of this management piece leverage MQTT between the devices for management.  Because of this we have a built in MQTT broker that allows for communication between Node-RED instance within the FlowFuse platform.  Lastly, with the release of Dashboards 2.0 from FlowFuse we have introduced the next generation of visualizations for the Node-RED platform.
+
+
+## Optimizing IoT with the MIND Stack
+
+It often makes sense to deploy a full MING stack, but in some deployments, it might be necessary to deploy a more simplistic version of the MING stack.  This is where MIND comes into play.
+
+
+### FlowFuse MQTT: Simplified Communications
+
+FlowFuse has a built-in feature called projects, which leverages MQTT, that allows the communication of data between FlowFuse runtimes of Node-RED.  One caveat is that this MQTT broker is only available within the FlowFuse platform. What this means is there needs to be some form of translation to be done within Node-RED.  This isn’t a big deal for small deployments, because often Node-RED runtimes at the edger are collecting data from various sources that aren’t MQTT.  The flow of data is as follows: 
+
+Sensor > Node-RED(FlowFuse Device Agent) > MQTT Encapsulated by FlowFuse > Node-RED(FlowFuse Platform) > InfluxDB
```

Comment:
@Yndira-E Could you make an artwork for this data flow?  I will make a ticket if it sounds reasonable.

Reply:
Hi @gdziuba sure, open an art request and I'll take care of it.

It would be great if you could provide some examples or even a description of what image, icon, or logo would represent each state of the flow. I'm not sure how "MQTT Encapsulated by FlowFuse" should look like, or the sensor. Is everything but the sensor and InfluxDB meant to be represented "inside" FlowFuse?

Reply:
https://github.com/FlowFuse/website/issues/2029

Reply:
I've just pushed it and added it in the .md file, please check it's ok.